### ISSUE TITLE
Possibilities for printing

### ISSUE THREAD

**Issue by ddevz** [aardappel/treesheets#169]
I thought treesheets seemed neat, so I started using it and I see a lot of great things with it.  However, now that I have a monstrous document, I need to make physical printouts for meetings.  Naturally it prints microscopically and can't be read.

I'll explain what I tried to do next:  So I made a top layer grid of cells, and call each cell a page.  I then zoom into cell one, print, zoom out, zoom into cell two, print, zoom out, forget to zoom into cell 3, print, realize it printed the whole thing, actually zoom into cell 3, print... you get the idea.

I see your already thinking about how to automatically break up the pages.  While your trying to figure that out, would it make sense to allow people to tag cells as "print this cell as a page".  Then "print pages" could print out just the marked cells.

Now, to "blue sky" it, some features could be:
 * Will print a cell marked for printing, even if it is contained by a cell that was already printed
 * Ability to tag cells as "print this cell as a page for scenario 1", or "print this cell as a page for scenario 2", and then options to print all scenario 1 pages, or print all scenario 2 pages.  If we do this, then it feels like people would never need to use more then one sheet and you'd just have a scenario for "meeting a", a scenario for "work stuff", a scenario for "personal stuff", etc.
 * Also, a maximum font size would be good, as a treesheet with just a few words in it comes out huge! :)

**Comment by aardappel** [comment]
Yeah I never seriously solved printing, as a) there is no natural way to split up a tree into pages, and I didn't think indicating this manually was very elegant, b) the printing functionality in wxWidgets (the API I go thru) appears buggy (rendering is not as accurate as on screen) and c) I never print anything.

You're basically asking to be able to mark certain cells as a root for printing, so the whole zoom in + print can be automated. That is pragmatic solution, I guess.

Another possibility would be for it to automatically try to find the right amount of stuff to fit on page.

**Comment by ddevz** [comment]
Yes, that's what I'm looking for.

As for "automatically trying to figure out how much to put on a page", it's often desirable to be able to show people just the part that pertains to them.  

I am imagining just having one big treesheet file.  When you use it, do you use one big treesheet file, or one file per topic?  If you do use one big file, how do you show the portion relating to just one topic to other people?

By the way, have you ever used emacs org-mode?  (If so I could make references to org-mode features.)

**Comment by aardappel** [comment]
I have some gigantic TreeSheets files, often with 1 column per topic area.. I find having it all in one file makes it quicker to overview "what I am working on" than in separate files.. Though it really depends on your working style. Whenever I share content, I use HTML export of the relevant zoomed in topic. HTML content is easy to paste into email, docs etc. I don't think I've ever used the print feature myself ;)

I've heard of emacs org-mode and once read an article on it, but no, not familiar with it. I don't use emacs.

**Comment by jeron1** [comment]
Hi, Reading this brought the HTML export function to my attention. This is a workable printing solution for me: HTML export, open in Firefox, print to PDF, print to paper. Many thanks.
